### PR TITLE
Maven transition

### PR DESCRIPTION
**Pull request by Michieldoesburg**
You can now run the visualizer through the GUI. The project has also switched to relying on maven for ASAT output instead of running the ASATs itself.

### CODE REVIEW

**Comment by ClintonCao** [comment]
Nice that the jar files etc. are removed for the ASATs. Now our repo is not filled with with a lot of of unnecessary files. 

One question though: Why did they test coverage dropped so dractically `(- 11.5 %)` ? A little bit more and it wouldn't have passed the build :stuck_out_tongue:

**Comment by Michieldoesburg** [comment]
When I run emma on my computer it says 86% coverage, why is coverall reporting 68%?

**Comment by ClintonCao** [comment]
Emma shows the coverage for both test and main package. Coverall only looks at the main package. 

If you only look at the main package, it should match with the result of Coverall

File: src/test/java/BlueTurtle/TSE/JavaControllerTest.java [hunk 1]
```diff
@@ -29,8 +29,6 @@ public void testUserDir() {
 	@Test
 	public void testAnalyserIsNullBeforeExecute() {
 		JavaController jc = new JavaController();
-		Analyser a = jc.getAnalyser();
-		assertNull(a);
```
Comment (ClintonCao) [inline src/test/java/BlueTurtle/TSE/JavaControllerTest.java]: Why not just remove this test?

**Comment by ClintonCao** [comment]
I've added a few more tests for the `JavaController`. I've also changed the coding style in some of the classes and fixed some of the warnings.

Though the test coverage is not as high as what we currently have on master, we need to know if the tool of SIG can be run on our repo. So I think we can merge this.

**Pull request merged**